sphinx/ext/imgmath.py

```diff
@@ -226,7 +237,9 @@ def render_math(
     latex = generate_latex_macro(image_format,
                                  math,
                                  self.builder.config,
-                                 self.builder.confdir)
+                                 self.builder.confdir,
```

Comment:
As `dvips` is the default state of `preview` when it detects that a `dvi` file is to be produced, and as it appears that `dvips` is needed work-around for usage with `xelatex` to let `preview` add PostScript information into the `dvi/xdv` that `dvisvgm` relies upon to recover the box dimension information, in particular the `depth`, I believe we can actually use always `dvips` here.  At same time, if we do that we make more difficult supporting a `pdf` intermediate file type, but for reasons at #6280 and for matters of evolving `dvisvgm` support of PDF, supporting only `dvi` and `xdv` looks reasonable.

Reply:
> As `dvips` is the default state of `preview` when it detects that a `dvi` file is to be produced,

Under the `tightpage` option.  This comes from this code in `prtightpage.def` file:
```
\ifnum\pr@graphicstype=\z@
  \ifcase
    \ifx\XeTeXversion\@undefined
      \ifx\pdfoutput\@undefined \@ne\fi
      \ifx\pdfoutput\relax \@ne\fi
      \ifnum\pdfoutput>\z@ \tw@\fi \@ne
    \else \thr@@\fi
  \or \ExecuteOptions{dvips}\relax
  \or \ExecuteOptions{pdftex}\relax
  \or \ExecuteOptions{xetex}\relax\fi\fi
```
I do need to check that `dvips` is not detrimental to `dvipng`.

Reply:
We do need to change the regex for depth with `png` if using `dvips` option.

Without:
```
$ dvipng -o ceaa1dfa0f274a525b952117e67d0cfd2de7d5ed.png -T tight -z9 -gamma 1.5 -D 110 -bg Transparent --depth math.dvi
This is dvipng 1.17 Copyright 2002-2015, 2019 Jan-Ake Larsson
[1 depth=1] 
```


With:
```
tmp0tx70t_4$ dvipng -o ceaa1dfa0f274a525b952117e67d0cfd2de7d5ed.png -T tight -z9 -gamma 1.5 -D 110 -bg Transparent --depth math.dvi
This is dvipng 1.17 Copyright 2002-2015, 2019 Jan-Ake Larsson
[1 (preview-latex version 13.1) depth=1] 
```

```diff
-depth_re = re.compile(r'\[\d+ depth=(-?\d+)\]')
+depth_re = re.compile(r'\[.* depth=(-?\d+)\]')
```

allows the depth information to be recovered if `dvipng` and `preview` was used with `dvips` option.

Reply:
In my testing the `png` files are identical whether or not using `dvips` option.  The only difference is that console output of `dvipng` is slightly changed as above.

So I think it is ok we use `dvips` always with `preview`.

I understand this is peripheral to your `tectonic` support objective, and that it may not be good style to introduce such a change on this occasion, but this looks to me like simplfying measure.

Reply:
Maybe we can use slightly more strict regex
```diff
-depth_re = re.compile(r'\[\d+ depth=(-?\d+)\]')
+depth_re = re.compile(r'\[\d.* depth=(-?\d+)\]')
```
but this does not look fragile anyhow.

Reply:
```diff
-        'tightpage': '' if image_format == 'png' else ',tightpage',
+        'tightpage': '' if image_format == 'png' else ',dvips,tightpage',
```
and don't add a `'dvips'` variable.

Reply:
I am a bit confused. My general impression is that you advocate to always include `dvips` as an option to the `preview` package, but in your last comment above, it seems to me that, you argue against it (maybe I am missing something). 

My two cents: I have verified empirically that your conclusions about the relation between the `dvips` option and `png` files is correct. I don't see the change of `depth_re` as very intrusive (even though it is not directly related to adding `tectonic`). But there is something else that bothers me - it is the intent. If we always add `dvips` as an option, then we have to complement this with a ton of comments (useful in such and such case, not useful in other cases, ...). While if we are explicit in the code and add it only where we use it, then it would be mostly clear for a person reading the code. 

Another thing. In the documentation of the `dvips` option (in the `preview` package) I read: "... This seriously hampers the usability of previews ...". While I am not entirely sure what this means, it seems to me that it is safer to omit this option whenever necessary (but I might be wrong). 

So I didn't change this in the current commit. But if you think that we should hard-code it after having read my comments above, then I would agree. In such a case, instead of

```python
\usepackage[active,dvips<%= tightpage %>]{preview}
```
I would opt for 
```python
\usepackage[active,<%= tightpage %>]{preview}
```
and adding `dvips,tightpage` upon instantiation of the template (as you suggest towards the end of your remarks). I prefer the latter option because, at least it excludes the `PNG` case.

Reply:
> I am a bit confused. My general impression is that you advocate to always include `dvips` as an option to the `preview` package, but in your last comment above, it seems to me that, you argue against it (maybe I am missing something).

Sorry about my very verbose comments.  I was initially vehemenly against `dvips` then after having understood it was required for `dvisvgm` to communicate to Sphinx the depth (as `dvisvgm` only looks for postscript specials, which `preview` writes only under `dvips` option), I started considering using it always for simplicity sake. The PostScript specials are ignored by `dvipng` (empirical).  With `latex`, the `tightpage` option automatically also triggers `dvips` option.  This is confirmed by this extract of the `preview` documentation on top of page 5:
> If neither dvips or pdftex have been specified, the corresponding option will get autodetected and invoked.

We must force it with `xetex` because `preview` in that case will not do `dvips` option.

About
> Another thing. In the documentation of the dvips option (in the preview package) I read: "... This seriously hampers the usability of previews ...". While I am not entirely sure what this means, it seems to me that it is safer to omit this option whenever necessary (but I might be wrong).

`preview` author refers to `dvips` used as class option or as option of some packages adding crop marks for example: the point is that `previews`'s `dvips` option is configured to counteract.  In LaTeX, a class option is also presented to packages.  So if `dvips` is used as class option, `preview` receives it too, and it does its counter-measures. I don't read PostScript so I can't be sure but it seems to be related to page dimensions. 

The whole point is that `dvisvgm` has a specific mechanism tied to recover the PostScript which `preview` adds to `dvi` (or `xdv`) under `tightpage` option.  But this mechanism is by default inihibited by `preview` when it is used with XeTeX (as you discovered, or with PDFTeX if used in PDF mode -- "latex" command is PDFTeX used in DVI mode, "pdflatex" Is PDFTeX used in PDF mode).

> I prefer the latter option because, at least it excludes the PNG case.

I agree.  Although `dvips` option seems not have any effect on `png` case, it is better not to introduce any change in a patch  which is about allowing `xelatex` as command and only regards `svg` case.